### PR TITLE
Update Mixer rule.selector to rule.match per design doc

### PR DESCRIPTION
**Pull request by mandarjog**
1. Update Mixer rule.selector to rule.match per design doc
2. Update comments.

### CODE REVIEW

File: mixer/v1/config/cfg.proto [hunk 1]
```diff
@@ -304,33 +304,29 @@ message EmailAddress {
 // instance constructed using the 'RequestCountByService' instance.
 //
 // ```yaml
-// actionRules:
-// - selector: destination.service == "*"
+// rule:
```
Comment (guptasu) [inline mixer/v1/config/cfg.proto]: this should be rules right since it is repeated ?

Comment (guptasu) [reply mixer/v1/config/cfg.proto]: Or every rule is different and pushed to config server as independent unit ?

Comment (guptasu) [reply mixer/v1/config/cfg.proto]: If that is the case then we need to remove this line completely and just start from 
match: ...
actions:

File: mixer/v1/config/cfg.proto [hunk 2]
```diff
@@ -304,33 +304,29 @@ message EmailAddress {
...
-  string selector = 1;
+  string match = 1;
 
-  // Optional. The actions that will be executed when selector evaluates to `true`.
+  // Optional. The actions that will be executed when match evaluates to `true`.
   repeated Action actions = 2;
-
-  // Optional. Nested action rules; their selectors are evaluated if this selector
-  // predicate evaluates to `true`.
-  repeated Rule rules = 3;
```
Comment (guptasu) [inline mixer/v1/config/cfg.proto]: nice

File: mixer/v1/config/cfg.proto (hunk 1 above)
Comment (mandarjog) [reply mixer/v1/config/cfg.proto]: ok I will remove that line.

**Pull request merged**